app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyChoiceQuestionScreen.kt

```diff
@@ -29,12 +27,25 @@ import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.kks.nimblesurveyjetpackcompose.R
 import com.kks.nimblesurveyjetpackcompose.model.SurveyAnswer
+import com.kks.nimblesurveyjetpackcompose.model.SurveyQuestionPickType
 import com.kks.nimblesurveyjetpackcompose.ui.theme.NeuzeitFamily
 import com.kks.nimblesurveyjetpackcompose.ui.theme.White50
 
+private const val CHOICE_WEIGHT = .75f
+
 @Composable
-fun SurveyChoiceQuestionScreen(answers: List<SurveyAnswer>, onChooseAnswer: (answers: List<SurveyAnswer>) -> Unit) {
-    var selectedIndexList by remember { mutableStateOf(emptyList<Int>()) }
+fun SurveyChoiceQuestionScreen(
+    answers: List<SurveyAnswer>,
+    pickType: SurveyQuestionPickType,
+    onChooseAnswer: (answers: List<SurveyAnswer>) -> Unit
+) {
+    val selectedIndexList = remember {
+        mutableStateListOf<Int>().also { snapShot ->
+            answers.forEachIndexed { index, surveyAnswer ->
+                if (surveyAnswer.selected) snapShot.add(index)
+            }
+        }
```

Comment:
We could simplify it.
```suggestion
        answers.filter { it.selected }.mapIndexed { index, _ -> index }.toMutableList()
```

Reply:
@AVI5HEK The result would be wrong with your approach. I want to get the actual indices of the selected in the actual list to restore the data. With your approach, result indices would be the indices of the filtered list but not the actual list.

<img width="888" alt="Screen Shot 2022-10-17 at 4 24 58 PM" src="https://user-images.githubusercontent.com/32578035/196141860-04b66baf-c95f-49cc-8831-ed7e64c71291.png">